### PR TITLE
Add Nightly benchmark workflow

### PR DESCRIPTION
**Pull request by varun-sundar-rabindranath**
SUMMARY:
Add a benchmarking workflow to run Nightly

TEST PLAN:
Manual benchmark triggers on AWS instances - check https://github.com/neuralmagic/nm-vllm/pull/66

### CODE REVIEW

**Comment by varun-sundar-rabindranath** [comment]
PR https://github.com/neuralmagic/neuralmagic-vllm/pull/66 has some time-saving changes - it'd be good to land that before this.

File: .github/workflows/nm-benchmark.yml [hunk 1]
```diff
@@ -11,7 +36,7 @@ on:
         type: string
         required: true
       timeout:
-        description: "approximate number of minutes to keep instance up (should be at least 20)."
+        description: "approximate number of minutes to keep instance up."
```
Comment (andy-neuma) [inline .github/workflows/nm-benchmark.yml:39]: maybe something like this, "maximum time runner will stay up"

File: .github/workflows/nm-benchmark.yml [hunk 2]
```diff
@@ -79,21 +105,4 @@ jobs:
...
-          S=$((R * 60))
-          echo "pausing for, ${R} minutes"
-
-      - name: pause workflow
-        run: |
-          M=${{ inputs.timeout }}
-          R=$((M - 15))
-          S=$((R * 60))
-          sleep $S
+          retention-days: 90
```
Comment (andy-neuma) [inline .github/workflows/nm-benchmark.yml:108]: maybe just 10 days for now

File: .github/workflows/nightly.yml [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+      # * is a special character in YAML so you have to quote this string
+      - cron: '0 0 * * *'
+
+    workflow_dispatch:
+
+jobs:
+
+    # multi-gpu
+    AWS-AVX2-192G-4-A10G-96G:
+        strategy:
```
Comment (andy-neuma) [inline .github/workflows/nightly.yml:14]: oh, we can likely just drop the "strategy" for now, since it only has one element.

**Review by andy-neuma — APPROVED** [review]
cool

**Pull request merged**